lib/autoproj/daemon/github_watcher.rb

```diff
@@ -105,6 +105,15 @@ def filter_events(events)
                 events.reject { |event| stale?(event) }
             end
 
+            def push_event_valid?(push_event)
+                branches = @client.branches(
+                    push_event.owner,
+                    push_event.name
+                )
+                branch_state = branches.find { |b| b.branch_name == push_event.branch }
+                return branch_state&.sha == push_event.head_sha
```

Comment:
Why the explicit return?

Reply:
VSCode is being so damn slow on my machine that I more often than not end up using vim ... hence the rubocop violations piling up :(

I don't understand how it became so bad.

Reply:
Have you tried disabling vscode-solargraph?

Reply:
Yes. Actually, I think it might be an issue with vscode-vim.

Reply:
Also had to disable the C++ extension.

Reply:
> Also had to disable the C++ extension.

This one is a huge memory hog if you a lot of packages. Intellisense caches all the symbols it finds and you end up with a very large database